clippy_lints/src/vec.rs

```diff
@@ -132,9 +132,20 @@ impl<'tcx> LateLintPass<'tcx> for UselessVec {
     fn check_crate_post(&mut self, cx: &LateContext<'tcx>) {
         for (span, lint_opt) in &self.span_to_lint_map {
             if let Some((hir_id, suggest_slice, snippet, applicability)) = lint_opt {
-                let help_msg = format!("you can use {} directly", suggest_slice.desc(),);
+                let help_msg = format!("you can use {} directly", suggest_slice.desc());
                 span_lint_hir_and_then(cx, USELESS_VEC, *hir_id, *span, "useless use of `vec!`", |diag| {
-                    diag.span_suggestion(*span, help_msg, snippet, *applicability);
+                    // If the `vec!` macro contains comment, better not emit any suggestion as it
+                    // would remove them. Let's just lint that the `vec!` is useless and let the user
+                    // decides what they wanna do.
```

Comment:
This comment is outdated:

```suggestion
                    // If the `vec!` macro contains comment, better not make the suggestion machine applicable as it
                    // would remove them.
```